app/src/main/java/ch/epfl/sdp/appart/database/firestoreservicehelpers/FirestoreImageHelper.java

```diff
@@ -25,14 +28,51 @@ public FirestoreImageHelper() {
 
     @NotNull
     @NonNull
-    public CompletableFuture<Boolean> putImage(Uri uri, String name, String path) {
-        if (uri == null || name == null) {
+    public CompletableFuture<Boolean> putImage(Uri uri, String imagePathAndName) {
+        if (uri == null || imagePathAndName == null) {
             throw new IllegalArgumentException("parameters cannot be null");
         }
         CompletableFuture<Boolean> isFinishedFuture = new CompletableFuture<>();
-        StorageReference fileReference = storage.getReference(path).child(name);
+        StorageReference fileReference = storage.getReference().child(imagePathAndName);
         fileReference.putFile(uri).addOnCompleteListener(
                 task -> isFinishedFuture.complete(task.isSuccessful()));
         return isFinishedFuture;
     }
+
+    /**
+     *  deletes an image from the Firebase Storage
+     * @param imagePathAndName this the complete path of the image (e.g. users/default/photo.jpeg)
+     */
+    @NotNull
+    @NonNull
+    public CompletableFuture<Boolean> deleteImage(String imagePathAndName) {
+        if (imagePathAndName == null) {
+            throw new IllegalArgumentException("deleteImage: parameters cannot be null");
+        }
+
+        CompletableFuture<Boolean> isFinishedFuture = new CompletableFuture<>();
+
+         /* The reason for this check is when a user has DEFAULT user icon (thus no image in storage)
+            The activity asks to delete previous profile image, which is not present, thus returns true directly */
+        if (imagePathAndName.contains("users/default/")) {
```

Comment:
it could be nice to have constants defined also for this

Reply:
Done!